src/backend/distributed/deparser/deparse_extension_stmts.c

```diff
@@ -0,0 +1,127 @@
+/*-------------------------------------------------------------------------
+ *
+ * deparse_extension_stmts.c
+ *	  All routines to deparse extension statements.
+ *	  This file contains deparse functions for extension statement deparsing
+ *    as well as related helper functions.
+ *
+ * Copyright (c), Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "distributed/deparser.h"
+#include "lib/stringinfo.h"
+#include "nodes/pg_list.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+
+/* Local functions forward declarations for helper functions */
+static void AppendCreateExtensionStmt(StringInfo str, CreateExtensionStmt *stmt);
+static void AppendDropExtensionStmt(StringInfo buf, DropStmt *stmt);
+static void AppendExtensionNameList(StringInfo buf, List *objects);
+
+
+/*
+ * DeparseCreateExtensionStmt builds and returns a string representing the
+ * CreateExtensionStmt to be sent to worker nodes.
+ */
+const char *
+DeparseCreateExtensionStmt(CreateExtensionStmt *createExtensionStmt)
+{
+	StringInfoData sql = { 0 };
+	initStringInfo(&sql);
+
+	AppendCreateExtensionStmt(&sql, createExtensionStmt);
+
+	return sql.data;
+}
+
+
+/*
+ * AppendCreateExtensionStmt appends a string representing the CreateExtensionStmt to a buffer
+ */
+static void
+AppendCreateExtensionStmt(StringInfo str, CreateExtensionStmt *createExtensionStmt)
+{
+	/*
+	 * Read required options from createExtensionStmt.
+	 * We do not fetch if_not_exist and cascade options as we will
+	 * append "IF NOT EXISTS" and "CASCADE" clauses regardless of
+	 * statement's content before propagating it to worker nodes.
+	 * We as also do not care old_version for now.
+	 */
+	const char *extensionName = createExtensionStmt->extname;
+
+	List *optionsList = createExtensionStmt->options;
+
+	const char *newVersion = GetCreateExtensionOption(optionsList, "new_version");
+	const char *schemaName = GetCreateExtensionOption(optionsList, "schema");
+
+	newVersion = quote_identifier(newVersion);
+	schemaName = quote_identifier(schemaName);
+
+	appendStringInfo(str,
+					 "CREATE EXTENSION IF NOT EXISTS %s WITH SCHEMA %s VERSION %s CASCADE",
```

Comment:
Not sure we should include a version, e.g. Citus would throw errors if the version that's installed on the workers is different (not uncommon).

Reply:
So, I guess we should only do if the user actually adds the `WITH version`? In that case, it probably still makes sense to send the version. If it fails, user should take action because he/she wants a specific version?

Reply:
Yes.

Reply:
I will make the necessary changes